include/quicr/encode.h

```diff
@@ -190,4 +190,35 @@ operator<<(MessageBuffer& msg, const Namespace& ns);
 MessageBuffer&
 operator>>(MessageBuffer& msg, Namespace& ns);
 
+
+/*===========================================================================*/
+// GET Message Types
+/*===========================================================================*/
+
+struct Get
+{
+  uint64_t transaction_id;
+  quicr::Namespace  resource; // resource to retrieve
+  // TODO - Add authz
+};
+
+MessageBuffer&
+operator<<(MessageBuffer& buffer, const Get& msg);
+MessageBuffer&
+operator>>(MessageBuffer& buffer, Get& msg);
+
+struct GetResponse
+{
+  quicr::Namespace resource;
+  SubscribeResult::SubscribeStatus response;
+  uint64_t transaction_id; // matches the get request
+  uintVar_t num_objects;
+  std::vector<PublishDatagram> objects;
```

Comment:
In reference to your TODO later, maybe this should be a vector of PublishStream, and you can add a Header field to this struct?